### PR TITLE
Add UART hardware flow control

### PR DESCRIPTION
**Pull request by d00616**
Adds hardware flow control to UART/UARTE. To enable RTS/CTS define PIN_SERIAL_CTS and PIN_SERIAL_RTS

### CODE REVIEW

**Comment by sandeepmistry** [comment]
From https://github.com/sandeepmistry/arduino-nRF5/pull/47#issuecomment-252699743

> Sending data from MCU to USB works without flow control enabled but in the other direction nothing is received.

@d00616 we must get this working before merging it that case.

**Comment by d00616** [comment]
> @d00616 we must get this working before merging it that case.
This is a PCA10000 board issue. Without implementing RTS/CTS only one direction is functional. With RTS/CTS enabled both directions are usable.

**Comment by sandeepmistry** [comment]
> With RTS/CTS enabled both directions are usable.

Just to confirm, with the patch can your sketch both receive data from the Arduino IDE serial monitor as well as send data?

**Comment by d00616** [comment]
> Just to confirm, with the patch can your sketch both receive data from the Arduino IDE serial monitor as well as send data?
Yes.

File: cores/nRF5/Uart.cpp [hunk 1]
```diff
@@ -188,8 +210,20 @@ size_t Uart::write(const uint8_t data)
 }
 
 #if defined(NRF52)
-Uart Serial( NRF_UART0, UARTE0_UART0_IRQn, PIN_SERIAL_RX, PIN_SERIAL_TX );
+  #define NRF_UART0_IRQn UARTE0_UART0_IRQn
+#elif defined(NRF51)
+  #define NRF_UART0_IRQn UART0_IRQn
+#endif
 
+#if defined(NRF_UART0_IRQn)
```
Comment (sandeepmistry) [inline cores/nRF5/Uart.cpp]: Is this check actually needed?

**Comment by d00616** [comment]
I have removed the "if defined(NRF_UART0_IRQn)" statement.

**Comment by sandeepmistry** [comment]
@d00616 thanks!

**Pull request merged**